node/cn/api.go

```diff
@@ -452,3 +433,82 @@ func (api *PrivateDebugAPI) getModifiedAccounts(startBlock, endBlock *types.Bloc
 	}
 	return dirty, nil
 }
+
+// getStartAndEndBlock returns start and end block based on the given startNum and endNum.
+func (api *PrivateDebugAPI) getStartAndEndBlock(startNum uint64, endNum *uint64) (*types.Block, *types.Block, error) {
+	var startBlock, endBlock *types.Block
+
+	startBlock = api.cn.blockchain.GetBlockByNumber(startNum)
+	if startBlock == nil {
+		return nil, nil, fmt.Errorf("start block number %d not found", startNum)
+	}
+
+	if endNum == nil {
+		endBlock = startBlock
+		startBlock = api.cn.blockchain.GetBlockByHash(startBlock.ParentHash())
+		if startBlock == nil {
+			return nil, nil, fmt.Errorf("block number %d has no parent", startNum)
+		}
+	} else {
+		endBlock = api.cn.blockchain.GetBlockByNumber(*endNum)
+		if endBlock == nil {
+			return nil, nil, fmt.Errorf("end block number %d not found", *endNum)
+		}
+	}
+
+	if startBlock.Number().Uint64() >= endBlock.Number().Uint64() {
+		return nil, nil, fmt.Errorf("start block height (%d) must be less than end block height (%d)", startBlock.Number().Uint64(), endBlock.Number().Uint64())
+	}
+
+	return startBlock, endBlock, nil
+}
+
+// GetModifiedStorageNodesByNumber returns all storage nodes of a contract account
+// that have been changed between the two blocks specified.
+//
+// With one parameter, returns the list of accounts modified in the specified block.
```

Comment:
Is it correct to return a list of accounts? In the sentence above, it says this function returns storage nodes

Reply:
You're right. I've missed to change the function comments. Please take a look =)